lib/utils.js

```diff
@@ -94,13 +92,34 @@ function parseParams(str) {
   return res;
 }
 
+const DecoderMap = new Map();
+
 function decodeText(text, textEncoding, destEncoding) {
-  var ret;
-  if (text && jsencoding.encodingExists(destEncoding)) {
-    try {
-      ret = jsencoding.TextDecoder(destEncoding)
-                      .decode(Buffer.from(text, textEncoding));
-    } catch(e) {}
+  let ret;
+  /**
+   * https://datatracker.ietf.org/doc/html/rfc7578#section-4.7
+   */
+  switch (destEncoding) {
+    case '7bit':
+      destEncoding = 'ascii';
```

Comment:
do we mutate arg instead of introducing new const for perf reasons?

Reply:
removed that